csrc/executor.h

```diff
@@ -125,7 +153,7 @@ class FusionExecutor : public NonCopyable {
     if (compiled_kernel_ != nullptr) {
       NVF_ERROR(compiled_kernel_->function != nullptr);
     }
-    return fusion_id_ != -1 && lowered_ && compiled_kernel_ != nullptr;
+    return !kernel_id_.empty() && lowered_ && compiled_kernel_ != nullptr;
```

Comment:
looks like `!kernel_id_.empty()` is the new check for a `legit compiled kernel`. Let's try to use this one instead of the arbitrary `fusion_id_ != -1` / `fusion_id_ > -1` / `fusion_id_ >= 0` all over the place.

Better yet, we can add an API for that